packages/relay-test-utils/RelayMockPayloadGenerator.js

```diff
@@ -391,32 +391,60 @@ class RelayMockPayloadGenerator {
     ) {
       value = defaultValues[applicationName];
 
-      // Let's validate the correctness of the provided enum value
-      // We will throw if value provided by mock resolvers is invalid
-      if (
-        enumValues !== null &&
-        !enumValues
-          .map(s => s.toUpperCase())
-          .includes(String(value).toUpperCase())
-      ) {
-        invariant(
-          false,
-          'RelayMockPayloadGenerator: Invalid value "%s" provided for enum ' +
-            'field "%s" via MockResolver.' +
-            'Expected one of the following values: %s.',
-          String(value),
-          `${path.join('.')}.${applicationName}`,
-          enumValues.map(v => `"${v}"`).join(', '),
-        );
+      if (Array.isArray(value) && enumValues !== null) {
```

Comment:
@sibelius let's have another iteration on this, and actually, move enumValues generation to a separate method. So, in the `mockScalar` we would call something like `value = this._getCorrectDefaulEnum(value, enumValues);`?

Reply:
it makes sense

I'm gonna modify this